website/components/LandingPage/OpenSource/index.js

```diff
@@ -43,73 +43,96 @@ function cacheGet(key) {
   }
 }
 
-function cacheSet(key, data) {
-  const twoMinutes = Date.now() + 2 * 60 * 1000;
+function cacheSet(key, data, length = 120000) {
+  const expires = Date.now() + length;
   try {
-    localStorage.setItem(
-      cacheKey(key),
-      JSON.stringify({ data, expires: twoMinutes })
-    );
+    localStorage.setItem(cacheKey(key), JSON.stringify({ data, expires }));
   } catch (error) {
     // Nothing.
   }
 }
 
-function cachedFetch(key, url, properties) {
+function cached(key, fallback, length) {
   const data = cacheGet(key);
   if (data) return Promise.resolve(data);
 
-  return fetch(url)
+  return fallback().then(data => {
+    cacheSet(key, data, length);
+    return data;
+  }, console.error);
+}
+
+function yyyymmdd(date) {
+  return date.toISOString().slice(0, 10);
+}
+
+function dateRanges(from, to) {
+  from = new Date(from);
+  to = new Date(to);
+
+  const dates = [yyyymmdd(from)];
+  while (from < to) {
+    dates.push(yyyymmdd(new Date(from.setMonth(from.getMonth() + 1))));
+  }
```

Comment:
Using `for` loop here would be simpler (and would get rid of these two reassignments above). Actually using `for` here could also get rid of the `reduce` below.